### PR TITLE
Skip python3 tests on python3.4 and below

### PR DESCRIPTION
**Pull request by sbc100**
We don't currently support python3.4 since it does support
circular dependencies in the way that we use them.  Currently
it generates:

```
Traceback (most recent call last):
  File "/b/build/slave/linux/build/src/src/work/wasm-install/emscripten/tools/cache.py", line 136, in <module>
    from . import shared
ImportError: cannot import name 'shared'
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  File "/b/build/slave/linux/build/src/src/work/wasm-install/emscripten/emcc.py", line 38, in <module>
    from tools import shared, jsrun, system_libs, client_mods, js_optimizer
  File "/b/build/slave/linux/build/src/src/work/wasm-install/emscripten/tools/shared.py", line 21, in <module>
    from . import jsrun, cache, tempfiles, colored_logger
  File "/b/build/slave/linux/build/src/src/work/wasm-install/emscripten/tools/cache.py", line 139, in <module>
    import shared
ImportError: No module named 'shared'
```

### CODE REVIEW

File: tests/test_other.py [hunk 1]
```diff
@@ -61,6 +61,19 @@ def decorated(self):
   return decorated
 
 
+def check_python3_version():
```
Comment (jgravelle-google) [inline tests/test_other.py]: I'd call this `is_python3_version_supported` or something.
Functions named "check" I expect to be void-returning, and signal their success/failure via exceptions or some other mechanism. Either that or they're ambiguously named (does `check_foo()` return true if there's something wrong with foo, or if foo is ok?)

File: tests/test_other.py [hunk 2]
```diff
@@ -61,6 +61,19 @@ def decorated(self):
...
+  """emscripten requires at least python3.5 since python3.4 and below do not
```
Comment (jgravelle-google) [inline tests/test_other.py]: This is a weird docstring. It's a good necessary comment, but my understanding is that docstrings are for explaining a function and how it's used, whereas this explains historical context.
In particular I'd expect it to say something like "returns true if the system version of python3 is supported" or something

File: tests/test_other.py [hunk 3]
```diff
@@ -61,6 +61,19 @@ def decorated(self):
...
+  support circular dependencies."""
+  python3 = Building.which('python3')
+  if not python3:
+    return False
+  output = run_process([python3, '--version'], stdout=PIPE).stdout
+  version = [int(x) for x in output.split(' ')[1].split('.')]
+  if version < [3, 5, 0]:
```
Comment (jgravelle-google) [inline tests/test_other.py]: `return version >= [3, 5, 0]`?
Also didn't know that comparison worked on lists, that's really elegant.

File: tools/shared.py [hunk 4]
```diff
@@ -1089,16 +1089,6 @@ def include_directive(paths):
 except NameError:
   pass
 
-# If we have 'env', we should use that to find python, because |python| may fail while |env python| may work
```
Comment (jgravelle-google) [inline tools/shared.py]: Should we not do this anymore? Why is this safe to remove?

Comment (sbc100) [reply tools/shared.py]: ENV_PREFIX is not used anywhere, I just noticed that while making this change.  I can separate if you prefer.

**Pull request merged**